src/main/java/org/folio/circulation/domain/UpdateRequestQueue.java

```diff
@@ -129,13 +149,50 @@ private CompletableFuture<Result<Request>> awaitPickup(Request request) {
             populateHoldShelfExpirationDate(
               request.withPickupServicePoint(servicePoint),
               tenantTimeZone
-            ))
+            ).map(calculatedRequest-> modifyHoldShelfExpirationDateBasedOnExpirationDateManagement(tenantTimeZone, calculatedRequest,
+              requestQueue, originalRequest)))
         );
     } else {
-      return completedFuture(succeeded(request));
+      Request updatedRequest = Request.from(request.asJson());
+      requestQueue.update(originalRequest, updatedRequest);
+
+      return requestRepository.update(request)
+        .thenComposeAsync(result -> result.after(v -> requestQueueRepository.updateRequestsWithChangedPositions(requestQueue)));
     }
   }
 
+  private RequestQueue modifyHoldShelfExpirationDateBasedOnExpirationDateManagement(ZoneId tenantTimeZone, Request calculatedRequest,
+                                                                                RequestQueue requestQueue, Request originalRequest) {
+
+    ExpirationDateManagement expirationDateManagement = calculatedRequest.getPickupServicePoint().getHoldShelfClosedLibraryDateManagement();
+    String intervalId = calculatedRequest.getPickupServicePoint().getHoldShelfExpiryPeriod().getIntervalId().toUpperCase();
+    //Old data where strategy is not set so default value but TimePeriod has MINUTES / HOURS
+    if(ExpirationDateManagement.KEEP_THE_CURRENT_DUE_DATE.name().equals(expirationDateManagement.name()) && checkShortTerm(intervalId)) {
```

Comment:
Need a space after `if` so it would be `if (ExpirationDate...`.

Reply:
Also think this expression can be simplified to `ExpirationDateManagement.KEEP_THE_CURRENT_DUE_DATE == expirationDateManagement && checkShortTerm(intervalId)`. Although they have a string representation they safely comparable with `==` and some would argue this is better for various reasons you can google on.

Reply:
I think there are a few reasons to prefer `==` when comparing enums in addition to simplifying the expression. I tend to agree with the most upvoted answers [here](https://stackoverflow.com/questions/1750435/comparing-java-enum-members-or-equals) for example. The compile time type checking is nice, as is the issue with null safety.

But using `equals` isn't wrong per se. However I do think comparing the result of `name()` can be removed so the expression can still be simplified to `ExpirationDateManagement.KEEP_THE_CURRENT_DUE_DATE.equals(expirationDateManagement)`.